grading/main_validator.cpp

```diff
@@ -435,12 +529,17 @@ int validateTestCases(const std::string &hw_id, const std::string &rcsid, int su
 
 
   // =======================================
-  // LOOP OVER ALL TEST CASES
-  nlohmann::json::iterator tc = config_json.find("testcases");
-  assert (tc != config_json.end());
-  for (unsigned int i = 0; i < tc->size(); i++) {
+  // Loop over all global testcases
+  std::vector<TestCase> testcases;
+  if (generate_all_output) {
+    testcases = getAllTestcases(config_json);
```

Comment:
oh, now I understand need for all vs for_submission.
but that should be in the comment at least, if not in the function naming
again, also think about merging into 1 function?
and pass in the generate output flag?